Some tests assume endianness

On [ppc64](https://koji.fedoraproject.org/koji/taskinfo?taskID=28589824) and [s390x](https://koji.fedoraproject.org/koji/taskinfo?taskID=28589826), two big-endian arches, two tests seem to fail due to some assumptions in tests and/or implementation:

```
_______________________ test_concatenate_types[dtypes1] ________________________
dtypes = (('<f4', '<f8'), '<f8')
    @pytest.mark.parametrize('dtypes', [(('>f8', '>f8'), '>f8'),
                                        (('<f4', '<f8'), '<f8')])
    def test_concatenate_types(dtypes):
        dts_in, dt_out = dtypes
        arrs = [np.zeros(4, dtype=dt) for dt in dts_in]
        darrs = [from_array(arr, chunks=(2,)) for arr in arrs]
    
        x = concatenate(darrs, axis=0)
>       assert x.dtype == dt_out
E       AssertionError: assert dtype('float64') == '<f8'
E        +  where dtype('float64') = dask.array<concatenate, shape=(8,), dtype=float64, chunksize=(2,)>.dtype
dask/array/tests/test_array_core.py:372: AssertionError

__________________________ test_concat_datetimeindex ___________________________
    def test_concat_datetimeindex():
        # https://github.com/dask/dask/issues/2932
        b2 = pd.DataFrame({'x': ['a']},
                          index=pd.DatetimeIndex(['2015-03-24 00:00:16'],
                                                 dtype='datetime64[ns]'))
        b3 = pd.DataFrame({'x': ['c']},
                          index=pd.DatetimeIndex(['2015-03-29 00:00:44'],
                                                 dtype='datetime64[ns]'))
    
        b2['x'] = b2.x.astype('category').cat.set_categories(['a', 'c'])
        b3['x'] = b3.x.astype('category').cat.set_categories(['a', 'c'])
    
        db2 = dd.from_pandas(b2, 1)
        db3 = dd.from_pandas(b3, 1)
    
        result = concat([b2.iloc[:0], b3.iloc[:0]])
>       assert result.index.dtype == '<M8[ns]'
E       AssertionError: assert dtype('>M8[ns]') == '<M8[ns]'
E        +  where dtype('>M8[ns]') = DatetimeIndex([], dtype='datetime64[ns]', freq=None).dtype
E        +    where DatetimeIndex([], dtype='datetime64[ns]', freq=None) = Empty DataFrame\nColumns: [x]\nIndex: [].index
dask/dataframe/tests/test_multi.py:1090: AssertionError
```
At least the latter looks like a test issue, but the former might be a problem in the implementation.

--- a/dask/array/tests/test_array_core.py
+++ b/dask/array/tests/test_array_core.py
@@ -363,7 +363,7 @@ def test_concatenate():
 
 
 @pytest.mark.parametrize('dtypes', [(('>f8', '>f8'), '>f8'),
-                                    (('<f4', '<f8'), '<f8')])
+                                    (('<f4', '<f8'), '=f8')])
 def test_concatenate_types(dtypes):
     dts_in, dt_out = dtypes
     arrs = [np.zeros(4, dtype=dt) for dt in dts_in]

